flask_sqlalchemy/__init__.py

```diff
@@ -512,6 +512,10 @@ def __init__(self, sa):
         self.sa = sa
 
     def __get__(self, obj, type):
+        if obj is not None and obj.__class__.__bases__[0].query is None:
```

Comment:
i8t should be noted that this is an api breaking changes as it sorts out both intentional calls (that really just miss a `type(...)` as well as accidental ones

it really should be a warning for a few releases